libminifi/src/utils/Id.cpp

```diff
@@ -215,6 +278,23 @@ void IdGenerator::initialize(const std::shared_ptr<Properties> & properties) {
   }
 }
 
+#ifndef WIN32
+bool IdGenerator::generateWithUuidImpl(unsigned int mode, UUID_FIELD output) {
+  void* uuid = nullptr;
+  try {
+    std::lock_guard<std::mutex> lock(uuid_mutex_);
+    uuid_impl_->make(mode);
+    uuid = uuid_impl_->binary();
+  } catch (...){
+    return false;
```

Comment:
As IdGenerator has a logger, I would appreciate logging in case we failed to generate unique ID.

Reply:
Fair point, will do.